src/mds/Server.cc

```diff
@@ -2893,9 +2893,13 @@ void Server::handle_client_open(MDRequestRef& mdr)
     return;
   }
 
-  // can only open a dir with mode FILE_MODE_PIN, at least for now.
-  if (cur->inode.is_dir())
+  if (!cur->inode.is_file()) {
+    // can only open non-regular inode with mode FILE_MODE_PIN, at least for now.
     cmode = CEPH_FILE_MODE_PIN;
+    // the inode is symlink and client wants to follow it, ignore the O_TRUNC flag.
+    if (cur->inode.is_symlink() && !(flags & O_NOFOLLOW))
+      flags &= ~O_TRUNC;
+  }
```

Comment:
I'm confused; the kernel will return EINVAL on a symlink truncate; this is making us just silently drop it. Where does the client follow the symlink and then truncate the target, and why would we want that behavior to be different from the kernel's?